---
title: "Reservoir Gradient"
author: "Jay T. Lennon, Megan L. Larsen, Nathan I. Wisnoski, & Mario E. Muscarella"
date: "`r format(Sys.time(), '%d %B, %Y')`"
header-includes:
   - \usepackage{array}
   - \usepackage{graphics}
output: 
  pdf_document:
    fig_caption: true
---

Effects of metacommunity processes on microbial assembly at the terrestrial-aquatic interface

# Initial Setup
```{r results='hide', message=FALSE}
rm(list=ls())
getwd()
setwd("./analyses")
opar <- par(no.readonly = TRUE)  # Saves plot defaults

# Import Required Packages
require("png")
require("grid")
require("reshape")
require("tidyverse")   
# require("akima")
# require("maps")
# require("rgdal")
# require("raster")
# require("OpenMx")
# require("ggmap")
# require("raster") 
# require("gridExtra")
require("vegan")
require("xtable")
require("viridis")
require("cowplot")
require("adespatial")
require("ggrepel")
require("gganimate")
require("maps")
require("rgdal")
require("iNEXT")

my.cols <- viridis(2, begin = .3, end = 0.8)

theme_set(theme_classic() + 
  theme(axis.title = element_text(size = 20),
        axis.title.x = element_text(margin = margin(t = 15, b = 15)),
        axis.title.y = element_text(margin = margin(l = 15, r = 15)),
        axis.text = element_text(size = 14),
        axis.text.x = element_text(margin = margin(t = 5)),
        axis.text.y = element_text(margin = margin(r = 5)),
        #axis.line.x = element_line(size = 1),
        #axis.line.y = element_line(size = 1),
        axis.line.x = element_blank(),
        axis.line.y = element_blank(),
        axis.ticks.x = element_line(size = 1),
        axis.ticks.y = element_line(size = 1),
        axis.ticks.length = unit(.1, "in"),
        panel.border = element_rect(color = "black", fill = NA, size = 1.5),
        legend.title = element_blank(),
        legend.text = element_text(size = 16)
        ))

```

## Load required R packages and tools
```{r results = 'hide'}
source("../bin/MothurTools.R")
se <- function(x, ...){sd(x, na.rm = TRUE)/sqrt(length(na.omit(x)))}
```

## Import Shared and Design Files
```{r}
# Define Inputs
# Design = general design file for experiment
# shared = OTU table from mothur with sequence similarity clustering
# Taxonomy = Taxonomic information for each OTU
design <- "../data/UL.design.txt"
shared <- "../data/UL.bac.final.shared"
taxon  <- "../data/UL.bac.final.0.03.taxonomy"

# Import Design
design <- read.delim(design, header=T, row.names=1)

# Import Shared Files
OTUs <- read.otu(shared = shared, cutoff = "0.03")    # 97% Similarity

# Import Taxonomy
OTU.tax <- read.tax(taxonomy = taxon, format = "rdp")

# Load environmental data
env.dat <- read.csv("../data/ResGrad_EnvDat.csv", header = TRUE)
env.dat <- env.dat[-16,]
```

## Data Transformations
```{r}
# Remove OTUs with less than two occurences across all sites
OTUs <- OTUs[, which(colSums(OTUs) >= 2)]

# Sequencing Coverage
coverage <- rowSums(OTUs)

# Good's Coverage
goods <- function(x = ""){
  1 - (sum(x == 1) / rowSums(x))
}
goods.c <- goods(OTUs)

# Remove Low Coverage Samples (This code removes two sites: Site 5DNA, Site 6cDNA)
lows <- which(coverage < 10000)
OTUs <- OTUs[-which(coverage < 10000), ]
design <- design[-which(coverage < 10000), ]
# Remove OTUs with less than two occurences across all sites
OTUs <- OTUs[, which(colSums(OTUs) >= 2)]
# OTUs <- rrarefy(OTUs, min(coverage))

# Make Relative Abundance Matrices
OTUsREL <- decostand(OTUs, method = "total")

# Log Transform Relative Abundances
OTUsREL.log <- decostand(OTUs, method = "log")
```

# Figure 1: Map of University Lake (reservoir transect)
```{r results='hide', fig.keep='none', warning=FALSE}
# ggplot theme
theme_maps <- function(base_size = 12, base_family = "Arial"){
  theme_bw(base_size = base_size, base_family = base_family) %+replace%
    theme(panel.background = element_rect(fill = "white", color = "black", size = 1),
          #panel.border = element_rect(color = "black"),
        #panel.margin = unit(1,1,1,1),
        panel.grid.major = element_blank(),
        panel.grid.minor = element_blank(),
        axis.ticks = element_blank(),
        axis.text = element_blank(),
        axis.title = element_blank(),
        legend.position = c(0.9,0.25)
        #,axis.title.x = element_blank(),axis.title.y = element_blank()
    )
}

# get shape files 
## for map inset
usa <- map_data("usa")
IN <- map_data("state", region = "Indiana")

## for main plot
ul <- readOGR("../maps","UniversityLakePoly")
#summary(ul) # Check projection and and datum
#ul <- spTransform(ul, CRS("+proj=longlat +datum=WGS84")) # transform if necessary
ul <- fortify(ul) # raster image for plotting with ggplot2

# If using google map as baselayer
#ul.coords <- c(lon = -86.503087, lat = 39.188686)
#ul.map <- get_map(location = ul.coords, 
#                  zoom = 17, #maptype = "terrain", 
#                  source = "google", color = "bw")
#base.map <- ggmap(ul.map, extent = "device", legend = "topleft")

# Main Map
main.map <- ggplot(aes(long,lat), data = env.dat) + 
  geom_polygon(fill = "grey", size = 0.5, color = 'black', data = ul, alpha = 1) +
  geom_point(size = 6, shape = 20) +
  labs(x = "Longitude (deg.)", y = "Latitude (deg.)") +
  # annotate("text", x = -86.5010, y = 39.18943, 
  #          label = "DAM", fontface = "bold") +
  annotate("text", x = -86.503, y = 39.1894, angle = 23, size = 6.8,
         label = "Soil %->% Riverine %->% Lacustrine %->% Dam", parse = TRUE) +
  scale_y_continuous(limits = c(39.1868, 39.190))
print(main.map)

# Inset Map
inset <- ggplot() +
  theme_maps() +
  theme(axis.text = element_blank(),
        axis.ticks.y = element_blank(),
        axis.ticks.x = element_blank(),
        panel.border = element_rect(color = "black", fill = NA, size = .25)) +
  labs(x = NULL, y = NULL)

inset <- inset + geom_map(data = usa, map = usa,
                    aes(x = long, y = lat, map_id = region),
                    color = "black", fill = "#e7e7e7", size = 0.15)

inset <- inset + geom_map(data = IN, map = IN,
                    aes(x = long, y=lat, map_id = region),
                    color = "black", fill = "black", size = 0.15)

inset <- inset + geom_point(aes(x = -86.503087, y = 39.188686), color = "white", shape = 18, size = 2)

print(inset)


# Start Plotting File
png(filename="../figures/01_map.png",
    width = 7, height = 6, units = "in", res = 400)

grid.newpage()
v1 <- viewport(width = 1, height = 1, x = 0.5, y = 0.5) #plot area for the main map
v2 <- viewport(width = 0.4, height = 0.3, x = 0.78, y = 0.33) #plot area for the inset map
print(main.map,vp = v1) 
print(inset,vp = v2)

dev.off() # this writes plot to folder
graphics.off() # shuts down open devices
img <- readPNG("../figures/01_map.png")
grid.raster(img)
```

# Analyze Diversity 
## Calculate Alpha Diversity
```{r}
# Observed Richness
S.obs <- rowSums((OTUs > 0) * 1)

# Simpson's Evenness
SimpE <- function(x = ""){
  x <- as.data.frame(x)
  D <- diversity(x, "inv")
  S <- sum((x > 0) * 1) 
  E <- (D)/S 
  return(E)
}
simpsE <- round(apply(OTUs, 1, SimpE), 3)
shan <- diversity(OTUs, index = "shannon")
exp.shan <- exp(shan)

# estimate asymptotic richness
divestim <- iNEXT(t(OTUs), datatype = "abundance", nboot = 999)
ggiNEXT(divestim) + 
  theme_minimal() +
  ggsave("../figures/rareextrapcurves.png", dpi = 300, 
         height = 10, width = 10, units = "in")

alpha.div <- cbind(design, S.obs, simpsE, shan, exp.shan)

divestim$iNextEst

hill.estim <- divestim$AsyEst %>% filter(Diversity == "Species richness") %>% 
  left_join(rownames_to_column(alpha.div), by = c("Observed" = "S.obs")) %>% 
  select(Site, rowname, station, molecule, type, distance) %>% 
  left_join(divestim$AsyEst, by = "Site")

hill.estim %>% filter(type == "water") %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = distance, y = Estimator, color = molecule, fill = molecule)) + 
  geom_point() + 
  geom_errorbar(aes(ymin = Estimator - s.e., ymax = Estimator + s.e.), 
                width = 8, alpha = 0.5) +
  geom_smooth(method = "lm") + 
  labs(x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m.)")),
       y = "Estimated Diversity (Hill Numbers)") +
  scale_color_manual(values = my.cols) +
  scale_fill_manual(values = my.cols) + 
  theme(legend.position = c(.89,.94),
        strip.text = element_text(size = 14),
        strip.background = element_blank()) +
  scale_x_reverse() +
  facet_grid(Diversity ~ ., scales = "free") + 
  ggsave("../figures/alpha-diversity.png", dpi = 500, width = 7, height = 8)

d <- estimateD(x = t(OTUs), datatype = "abundance", base = "coverage")

```

## Alpha Diversity Across Gradient
```{r}
alpha.div <- divestim$AsyEst %>% filter(Diversity == "Species richness") %>% 
  left_join(rownames_to_column(alpha.div), by = c("Observed" = "S.obs"))
# Seperate data based on lake and soil samples
lake <- alpha.div[alpha.div$type == "water",]
soil <- alpha.div[alpha.div$type == "soil", ]


# Calculate Linear Model
model.rich <- lm(lake$Estimator ~ lake$distance * lake$molecule)

# Calculate Confidance Intervals of Model
newdata.rich <- data.frame(cbind(lake$molecule, lake$distance))
conf95.rich <- predict(model.rich, newdata.rich, interval="confidence")

# Average Richess in Terrestrial Habitat
mean(soil$Estimator)

# Dummy Variables Regression Model ("Species Richness")
D1 <- (lake$molecule == "RNA")*1
fit.Fig.3a <- lm(lake$Estimator ~ lake$distance + D1 + lake$distance*D1)
summary(fit.Fig.3a)
D1.R2 <- round(summary(fit.Fig.3a)$r.squared, 2)

DNA.int.3a <- fit.Fig.3a$coefficients[1]
DNA.slp.3a <- fit.Fig.3a$coefficients[2]
RNA.int.3a <- DNA.int.3a + fit.Fig.3a$coefficients[3]
RNA.slp.3a <- DNA.slp.3a + fit.Fig.3a$coefficients[4]
```

## Figure 2: Estimated Richness
```{r}
lake %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = distance, y = Estimator, color = molecule, fill = molecule)) + 
  geom_point(size = 2, alpha = 0.8, show.legend = T) +
  geom_errorbar(aes(ymin = Estimator - s.e., ymax = Estimator + s.e.), width = 8, alpha = 0.5) +
  geom_smooth(method = 'lm', show.legend = T) +
  annotate(geom = "text", x = 50, y = 1800, size = 6,
           label = paste0("R^2== ",D1.R2), parse = T) +
  scale_x_reverse(limits = c(400,0)) + 
  labs(x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m.)")),
       y = "Estimated Richness") +
  theme(legend.position = c(0.9, 0.9)) +
  scale_color_manual(values = my.cols) +
  scale_fill_manual(values = my.cols) +
  scale_y_continuous(limits = c(0, 3000)) + 
  ggsave("../figures/02_Richness.pdf", bg = "white", width = 7, height = 6) + 
  ggsave("../figures/02_Richness.png", width = 7, height = 6, dpi = 500)
```

## Track structure across space
```{r}
ul.pcoa <- cmdscale(vegdist(OTUsREL.log, method="bray"), 2, eig = T, add = T)
explainvars <- round(eigenvals(ul.pcoa)[c(1,2)]/sum(eigenvals(ul.pcoa)),3) *100
water.pcvals <- data.frame(scores(ul.pcoa)) %>% 
  rownames_to_column("name") %>% 
  left_join(rownames_to_column(design, "name")) %>% 
  arrange(desc(distance)) %>% filter(type == "water")
pc_dists <- data_frame(
  DNA_dim1 = subset(water.pcvals, molecule == "DNA")$Dim1,
  DNA_dim2 = subset(water.pcvals, molecule == "DNA")$Dim2,
  RNA_dim1 = subset(water.pcvals, molecule == "RNA")$Dim1,
  RNA_dim2 = subset(water.pcvals, molecule == "RNA")$Dim2)
data.frame(scores(ul.pcoa)) %>% 
  rownames_to_column("name") %>% 
  left_join(rownames_to_column(design, "name")) %>% 
  arrange(desc(distance)) %>% filter(type == "water") %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = Dim1, y = Dim2)) +
  geom_point(alpha = 0.5, aes(color = molecule)) + 
  geom_path(alpha = 0.75, arrow = arrow(angle = 20,
                          length = unit(0.35, "cm"),
                          type = "closed"), aes(color = molecule)) +
  scale_color_manual("Community Subset", values = my.cols) +
  geom_segment(data = pc_dists,
               aes(x = DNA_dim1, y = DNA_dim2,
                   xend = RNA_dim1, yend = RNA_dim2),
               alpha = 0.05) +
  coord_fixed() +
  labs(x = paste0("PCoA 1 (", explainvars[1],"%)"),
       y = paste0("PCoA 2 (", explainvars[2],"%)")) +
  ggsave("../figures/active-tot-pcoa-trajectories.pdf", bg = "white", width = 8, height = 8) +
  ggsave("../figures/active-tot-pcoa-trajectories.png", width = 8, height = 8)

# animation
traj.animate <- data.frame(scores(ul.pcoa)) %>% 
  rownames_to_column("name") %>% 
  left_join(rownames_to_column(design, "name")) %>% 
  arrange(desc(distance)) %>% filter(type == "water", station != "UL17", station != "UL18") %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  mutate(transect = desc(distance)) %>% 
  ggplot(aes(x = Dim1, y = Dim2, frame = transect, cumulative = TRUE)) +
  geom_point(alpha = 0.75, size = 10, aes(color = molecule)) + 
  geom_path(alpha = 1, size = 1, arrow = arrow(angle = 20,
                          length = unit(0.35, "cm"),
                          type = "closed"), aes(color = molecule)) +
  scale_color_manual(values = my.cols) +
  theme(legend.title = element_blank()) +
  coord_fixed() +
  labs(x = paste0("PCoA 1 (", explainvars[1],"%)"),
       y = paste0("PCoA 2 (", explainvars[2],"%)"))

gganimate(traj.animate, filename = "../figures/trajectory-animation.mp4", ani.width = 800, ani.height = 800)
```

## Similarity To Terrestrial Habitat Across Gradient (Terrestrial Influence)
```{r}
# Similarity to Soil Sample
UL.bray      <- 1-as.matrix(vegdist(OTUsREL.log, method="bray"))
UL.bray.lake <- UL.bray[-c(1:3), 1:3] 
bray.mean    <- round(apply(UL.bray.lake, 1, mean), 3)
bray.se      <- round(apply(UL.bray.lake, 1, se), 3)
UL.sim       <- cbind(design[-c(1:3), ], bray.mean, bray.se)

# Calculate Linear Model
model.terr <- lm(UL.sim$bray.mean ~ UL.sim$distance * UL.sim$molecule)

# Calculate Confidance Intervals of Model
newdata.terr <- data.frame(cbind(UL.sim$molecule, UL.sim$distance))
conf95.terr <- predict(model.terr, newdata.terr, interval="confidence")

# Dummy Variables Regression Model ("Terrestrial Influence")
D2 <- (UL.sim$molecule == "RNA")*1
fit.Fig.3b <- lm(UL.sim$bray.mean ~ UL.sim$distance + D2 + UL.sim$distance*D2)
D2.R2 <- round(summary(fit.Fig.3b)$r.squared, 2)
summary(fit.Fig.3b)

DNA.int.3b <- fit.Fig.3b$coefficients[1]
DNA.slp.3b <- fit.Fig.3b$coefficients[2]
RNA.int.3b <- DNA.int.3b + fit.Fig.3b$coefficients[3]
RNA.slp.3b <- DNA.slp.3b + fit.Fig.3b$coefficients[4]
```

## Figure 3: Similarity to Soils
```{r}
UL.sim %>% 
  mutate(molecule = ifelse(UL.sim$molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = distance, y = bray.mean, color = molecule, fill = molecule)) +
  geom_point(alpha = 0.8, show.legend = T) + 
  geom_smooth(method = "lm", show.legend = T) + 
  labs(y = "Similarity to Soil Community", 
       x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m.)"))) + 
  scale_color_manual(values = my.cols) + 
  scale_fill_manual(values = my.cols) +
  theme(legend.position = c(0.9, 0.9)) +
  scale_x_reverse(limits = c(400,0)) + 
  annotate(geom = "text", x = 50, y = 0.15, size = 6,
           label = paste0("R^2== ",round(summary(fit.Fig.3b)$r.squared, 2)), parse = T) +
  ggsave("../figures/03_similarity-to-soils.pdf", bg = "white", height = 6, width = 7) +
  ggsave("../figures/03_similarity-to-soils.png", dpi = 500, height = 6, width = 7)
```

## Similarity To Lake Habitat Across Gradient
```{r}
# Similarity to Lake Samples 1 and 2
UL.bray2      <- 1 - as.matrix(vegdist(OTUsREL.log, method="bray"))
UL.bray.lake2 <- UL.bray[-c(1:3), 4:7]
UL.sim2       <- cbind(design[-c(1:3), ], 
                       "DNA" = apply(UL.bray.lake2[,c(1,3)], 1, mean), 
                       "RNA" = apply(UL.bray.lake2[,c(2,4)], 1, mean))

UL.sim2 %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = distance, y = DNA, color = molecule, fill = molecule)) +
  geom_point() + 
  geom_smooth(method = "lm") + 
  labs(y = "Similarity to total lake community (DNA)", 
       x = (expression("Soil" %->% "Stream" %->% "Reservoir" %->% "Dam Transect"))) +
  scale_x_reverse() +
  scale_color_manual("Community Subset", values = my.cols) +
  scale_fill_manual("Community Subset", values = my.cols)

UL.sim2 %>% 
  mutate(molecule = ifelse(molecule == "DNA", "Total", "Active")) %>% 
  ggplot(aes(x = distance, y = RNA, color = molecule, fill = molecule)) +
  geom_point() + 
  geom_smooth(method = "lm") + 
  labs(y = "Similarity to active lake community (RNA)", 
       x = (expression("Soil" %->% "Stream" %->% "Reservoir" %->% "Dam Transect"))) +
  scale_x_reverse() +
  scale_color_manual("Community Subset", values = my.cols) +
  scale_fill_manual("Community Subset", values = my.cols)


# Calculate Linear Model
model.lake1 <- lm(UL.sim2$DNA ~ UL.sim2$distance * UL.sim2$molecule)
model.lake2 <- lm(UL.sim2$RNA ~ UL.sim2$distance * UL.sim2$molecule)
summary(model.lake1)
summary(model.lake2)

# Calculate Confidance Intervals of Model
newdata.lake <- data.frame(cbind(UL.sim2$molecule, UL.sim2$distance))
conf95.lake <- predict(model.lake1, newdata.lake, interval="confidence")

# Dummy Variables Regression Model ("Lake Influence")
D3 <- (UL.sim2$molecule == "RNA")*1
fit.Fig.3c <- lm(UL.sim2$DNA ~ UL.sim2$distance + D3 + UL.sim2$distance*D3)
# summary(fit.Fig.3c)

DNA.int.3c <- fit.Fig.3c$coefficients[1]
DNA.slp.3c <- fit.Fig.3c$coefficients[2]
RNA.int.3c <- DNA.int.3c + fit.Fig.3c$coefficients[3]
RNA.slp.3c <- DNA.slp.3c + fit.Fig.3c$coefficients[4]
```

# Identifying the Soil Bacteria
```{r}
soil.only <- OTUs[, which(colSums(OTUs[-c(1:3),]) == 0)] # what is present only in soil
lake.n.soil <- OTUs[, setdiff(colnames(OTUs),colnames(soil.only))] # what is present across all samples, but not just in soils


# separate lake and soil samples
lake.total <- OTUs[which(design$molecule == "DNA", design$type == "water"),]
soil.total <- OTUs[which(design$molecule == "DNA", design$type == "soil"),]

# which otus are present in both lake and soil samples
lake.and.soil.total <- OTUs[which(design$molecule == "DNA", design$type == "water"),
                            which(colSums(lake.total) > 0 & colSums(soil.total) > 0)]

# isolate just the dna and rna lake communities
w.dna <- OTUs[which(design$molecule == "DNA" & design$type == "water"), ]
w.rna <- OTUs[which(design$molecule == "RNA" & design$type == "water"), ]

# pull out the lake rna counds for otus found in lake and soil
lake.and.soil.act <- w.rna[,colnames(lake.and.soil.total)]

# of these lake and soil taxa, which are never active?
nvr.act <- which(colSums(lake.and.soil.act) == 0)

# pull out their dna abundances and calculate richness
terr.lake <- w.dna[ , c(names(nvr.act))]
terr.rich <- rowSums((terr.lake > 0) * 1)
terr.REL <- rowSums(terr.lake) / rowSums(w.dna) 
design.dna <- design[which(design$molecule == "DNA" & design$type == "water"), ]
terr.rich.log <- log10(terr.rich)
terr.REL.log <- log10(terr.REL)

terr.mod1 <- lm(terr.rich.log ~ design.dna$distance)
summary(terr.mod1)
T1.R2 <- round(summary(terr.mod1)$r.squared, 2)
T1.int <- terr.mod1$coefficients[1]
T1.slp <- terr.mod1$coefficients[2]

terr.mod2 <- lm(terr.REL.log ~ design.dna$distance)
summary(terr.mod2)
T2.R2 <- round(summary(terr.mod2)$r.squared, 2)
T2.int <- terr.mod2$coefficients[1]
T2.slp <- terr.mod2$coefficients[2]
```

## Figure 4: Transient decay
```{r}
data_frame(transient_rich = terr.rich, distance = design.dna$distance) %>% 
  ggplot(aes(x = distance, y = transient_rich)) + 
  geom_point(alpha = 0.8, color = "chocolate4") + 
  geom_smooth(method = "lm", color = "chocolate4", fill = "chocolate4") +
  scale_x_reverse(limits = c(400,0)) +
  scale_y_log10() +
  annotation_logticks(sides = "l") +
  labs(x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m.)")),
       y = "Soil-Derived Transients") +
  annotate("text", x = 50, y = 750, size = 6, label = paste0("R^2== ",T1.R2), parse = T) +
  ggsave("../figures/04_transient-rich.pdf", bg = "white", height = 6, width = 7) +
  ggsave("../figures/04_transient-rich.png", dpi = 500, height = 6, width = 7)

```

# Define Core Lake Taxa
```{r}
# identify otus in soil samples and lake samples
in.soil <- OTUs[, which(colSums(OTUs[c(1:3),]) > 0 )]
in.lake <- OTUs[, which(colSums(OTUs[-c(1:3),]) > 0)]

# isolate just the rna water samples and convert to presence-absence
in.lake.rna <- in.lake[which(design$molecule == "RNA" & design$type == "water"), ]
in.lake.rna.pa <- (in.lake.rna > 0) * 1

# define the 'core' taxa as otus present in 90% of samples
in.lake.core <- w.dna[, which((colSums(in.lake.rna.pa) / nrow(in.lake.rna.pa)) >= 0.8)]

# of the core, how many are also in the soil samples?
in.lake.core.from.soils <- in.lake.core[, intersect(colnames(in.lake.core), colnames(in.soil))]

# of the core which are not in the soil samples
in.lake.core.not.soils <- in.lake.core[, setdiff(colnames(in.lake.core), colnames(in.soil))]


in.lake.core.from.soils.REL <- in.lake.core.from.soils / rowSums(w.dna)
in.soil.plot <- as.data.frame(in.lake.core.from.soils.REL) %>% 
  rownames_to_column("sample_ID") %>% 
  gather(otu_id, rel_abundance, -sample_ID) %>% 
  left_join(rownames_to_column(design.dna, "sample_ID")) %>% 
  ggplot(aes(x = distance, y = rel_abundance, color = otu_id)) + 
  labs(title = "OTUs first found in soil") +
  geom_line(alpha = 0.25, stat = "smooth", method = "lm", se = F, show.legend = F)

in.lake.core.not.soils.REL <- in.lake.core.not.soils / rowSums(w.dna) 
in.lake.plot <- as.data.frame(in.lake.core.not.soils.REL) %>% 
  rownames_to_column("sample_ID") %>% 
  gather(OTU, rel_abundance, -sample_ID) %>% 
  left_join(rownames_to_column(design.dna, "sample_ID")) %>% 
  left_join(OTU.tax) %>% 
  ggplot(aes(x = distance, y = rel_abundance, color = OTU)) + 
  geom_line(alpha = 0.25, stat = "smooth", method = "lm", se = F, show.legend = F) +
  labs(title = "OTUs first found in the reservoir")

plot_grid(in.soil.plot + theme(axis.title.x = element_blank()), in.lake.plot, align = "hv", ncol = 1) + ggsave("../figures/coretaxa.png", height = 10, width = 6)


data.frame(mean_relabund = colMeans(in.lake.core.from.soils.REL)) %>% 
  rownames_to_column(var = "OTU") %>% left_join(OTU.tax) %>% 
  mutate(Taxon = paste(Phylum, Class, Order)) %>% 
  arrange(desc(mean_relabund)) %>% 
  ggplot() + 
  geom_bar(aes(x = Taxon, y = mean_relabund), stat = "identity") +
  coord_flip()
soil.core.mods <- apply(in.lake.core.from.soils.REL, MARGIN = 2, 
    FUN = function(x) summary(lm(x ~ design.dna$distance))$coefficients[2,c(1,4)])
rownames(soil.core.mods) <- c("slope", "pval")
soil.core.decresing <- as.data.frame(t(soil.core.mods)) %>% 
  rownames_to_column("OTU") %>% 
  filter(pval < 0.05 & slope > 0) %>%   # rel abund decreases toward dam
  left_join(OTU.tax)
soil.core.increasing <- as.data.frame(t(soil.core.mods)) %>% 
  rownames_to_column("OTU") %>% 
  filter(pval < 0.05 & slope < 0) %>%   # rel abund increases toward dam
  left_join(OTU.tax)

data.frame(mean_relabund = colMeans(in.lake.core.not.soils.REL)) %>% 
  rownames_to_column(var = "OTU") %>% left_join(OTU.tax) %>% 
  mutate(Taxon = paste(Phylum, Class, Order)) %>% 
  arrange(desc(mean_relabund)) %>% 
  ggplot() + 
  geom_bar(aes(x = Taxon, y = mean_relabund), stat = "identity") +
  coord_flip()
nonsoil.core.mods <- apply(in.lake.core.not.soils.REL, MARGIN = 2, FUN = function(x) summary(lm(x ~ design.dna$distance))$coefficients[2,c(1,4)])
rownames(nonsoil.core.mods) <- c("slope", "pval")
nonsoil.core.decreasing <- as.data.frame(t(nonsoil.core.mods)) %>% 
  rownames_to_column("OTU") %>% 
  filter(pval < 0.05 & slope > 0) %>%   # rel abund decreases toward dam
  left_join(OTU.tax)
nonsoil.core.increasing <- as.data.frame(t(nonsoil.core.mods)) %>% 
  rownames_to_column("OTU") %>% 
  filter(pval < 0.05 & slope < 0) %>%   # rel abund increases toward dam
  left_join(OTU.tax)

as.data.frame(OTUsREL[,nonsoil.core.increasing$OTU]) %>% rownames_to_column("sampleID") %>% 
  left_join(rownames_to_column(design, "sampleID")) %>% 
  gather(OTU, rel_abund, -station, -molecule, -type, -distance, -sampleID) %>% 
  filter(molecule == "DNA") %>% left_join(OTU.tax) %>% 
  mutate(taxon = paste(Class, Order, Family)) %>% 
  ggplot(aes(x = distance, y = rel_abund, color = taxon)) + 
  geom_point() + 
  geom_smooth(method = "lm") + 
  scale_x_reverse()

as.data.frame(OTUsREL[,soil.core.increasing$OTU]) %>% rownames_to_column("sampleID") %>% 
  left_join(rownames_to_column(design, "sampleID")) %>% 
  gather(OTU, rel_abund, -station, -molecule, -type, -distance, -sampleID) %>% 
  filter(molecule == "DNA") %>% left_join(OTU.tax) %>% 
  mutate(taxon = paste(Class, Order, Family)) %>% 
  ggplot(aes(x = distance, y = rel_abund, color = taxon)) + 
  geom_point() + 
  geom_smooth(method = "lm") +
  scale_x_reverse()


# how much do the different core components contribute to total abundances
in.lake.core.soil.REL <- rowSums(in.lake.core.from.soils) / rowSums(w.dna)
in.lake.core.water.REL <- rowSums(in.lake.core.not.soils) / rowSums(w.dna)
```


## Taxonomic Analysis
```{r, results="asis"}
# Taxa comprising total lake 'core', those from soils, and those not from soils
core.taxa <- OTU.tax[OTU.tax$OTU %in% colnames(in.lake.core),]

core.soil.taxa <- OTU.tax[OTU.tax$OTU %in% colnames(in.lake.core.from.soils),]
core.water.taxa <- OTU.tax[OTU.tax$OTU %in% colnames(in.lake.core.not.soils),]

# Get relative abundances for each of the core taxa 
core.soil.taxa.DNA.REL <- OTUsREL[which(design$molecule == "DNA" & design$type == "water"),
                                  as.numeric(rownames(core.soil.taxa))]
core.water.taxa.DNA.REL <- OTUsREL[which(design$molecule == "DNA" & design$type == "water"),
                                   as.numeric(rownames(core.water.taxa))]
core.soil.taxa.RNA.REL <- OTUsREL[which(design$molecule == "RNA" & design$type == "water"),
                                  as.numeric(rownames(core.soil.taxa))]
core.water.taxa.RNA.REL <- OTUsREL[which(design$molecule == "RNA" & design$type == "water"),
                                   as.numeric(rownames(core.water.taxa))]

core.soil.taxa.DNA.REL.max <- as.matrix(apply(core.soil.taxa.DNA.REL, 2, max))
core.soil.taxa.RNA.REL.max <- as.matrix(apply(core.soil.taxa.RNA.REL, 2, max))
core.water.taxa.DNA.REL.max <- as.matrix(apply(core.water.taxa.DNA.REL, 2, max))
core.water.taxa.RNA.REL.max <- as.matrix(apply(core.water.taxa.RNA.REL, 2, max))

core.soil.taxa.DNA.REL.min <- as.matrix(apply(core.soil.taxa.DNA.REL, 2, min))
core.soil.taxa.RNA.REL.min <- as.matrix(apply(core.soil.taxa.RNA.REL, 2, min))
core.water.taxa.DNA.REL.min <- as.matrix(apply(core.water.taxa.DNA.REL, 2, min))
core.water.taxa.RNA.REL.min <- as.matrix(apply(core.water.taxa.RNA.REL, 2, min))

core.soil.taxa.DNA.REL.mean <- as.matrix(apply(core.soil.taxa.DNA.REL, 2, mean))
core.soil.taxa.RNA.REL.mean <- as.matrix(apply(core.soil.taxa.RNA.REL, 2, mean))
core.water.taxa.DNA.REL.mean <- as.matrix(apply(core.water.taxa.DNA.REL, 2, mean))
core.water.taxa.RNA.REL.mean <- as.matrix(apply(core.water.taxa.RNA.REL, 2, mean))

core.soil.taxa.DNA.REL.bounds <- cbind(core.soil.taxa.DNA.REL.min, core.soil.taxa.DNA.REL.max,
                                       core.soil.taxa.RNA.REL.min, core.soil.taxa.RNA.REL.max,
                                       core.soil.taxa.DNA.REL.mean, core.soil.taxa.RNA.REL.mean)
colnames(core.soil.taxa.DNA.REL.bounds) <- c("DNA.min", "DNA.max", "RNA.min", "RNA.max", "DNA.mean", "RNA.mean")

core.water.taxa.DNA.REL.bounds <- cbind(core.water.taxa.DNA.REL.min, core.water.taxa.DNA.REL.max,
                                       core.water.taxa.RNA.REL.min, core.water.taxa.RNA.REL.max,
                                       core.water.taxa.DNA.REL.mean, core.water.taxa.RNA.REL.mean)
colnames(core.water.taxa.DNA.REL.bounds) <- c("DNA.min", "DNA.max", "RNA.min", "RNA.max", "DNA.mean", "RNA.mean")

# core.soil and core.water are summaries of lake core
core.soil <- as.data.frame(cbind(core.soil.taxa$Family, core.soil.taxa$Genus,
                                 round(core.soil.taxa.DNA.REL.bounds[,1:4], 2)))
colnames(core.soil) <- c("Family", "Genus", "DNA.min", "DNA.max", "RNA.min", "RNA.max")
core.water <- as.data.frame(cbind(core.water.taxa$Family, core.water.taxa$Genus, 
                                  round(core.water.taxa.DNA.REL.bounds[,1:4],2)))
colnames(core.water) <- c("Family", "Genus", "DNA.min", "DNA.max", "RNA.min", "RNA.max")

# Core Soil LaTeX Table
addtorow <- list()
addtorow$pos <- list(0, 0)
addtorow$command <- c("& \\multicolumn{1}{c}{Class} & \\multicolumn{1}{c}{Order} & 
                      \\multicolumn{2}{c}{DNA} & \\multicolumn{2}{c}{RNA} \\\\\n", 
                      "& &  & min & max & min & max \\\\\n")
core.soil.tab <- xtable(core.soil)
align(core.soil.tab) <- "crrrrrr"
print(core.soil.tab, add.to.row = addtorow, include.colnames = FALSE, 
      type= "latex", file="../tables/table1.tex")
print(core.soil.tab, add.to.row = addtorow, include.colnames = FALSE, comment = FALSE)

core.water.tab <- xtable(core.water)
align(core.water.tab) <- "crrrrrr"
print(core.water.tab, add.to.row = addtorow, include.colnames = FALSE, 
      type= "latex", file="../tables/table2.tex")
print(core.water.tab, add.to.row = addtorow, include.colnames = FALSE, comment = FALSE)

high.activity.soil.core <- as.data.frame(core.soil.taxa.DNA.REL.bounds) %>%
  rownames_to_column("OTU") %>% 
  filter(RNA.max > 0) %>% arrange(desc(RNA.max)) %>% 
  left_join(OTU.tax)
high.activity.water.core <- as.data.frame(core.water.taxa.DNA.REL.bounds) %>%
  rownames_to_column("OTU") %>% 
  filter(RNA.max > 0) %>% arrange(desc(RNA.max)) %>% 
  left_join(OTU.tax)

mean.soil.abunds.soil.core <- OTUsREL[which(design$type == "soil"), high.activity.soil.core$OTU] %>% 
  colMeans %>% data.frame(mean_soil_relabund = .) %>% 
  rownames_to_column("OTU") %>% arrange(desc(mean_soil_relabund))
max.soil.abunds.soil.core <- OTUsREL[which(design$type == "soil"), high.activity.soil.core$OTU] %>% 
  apply(X = ., MARGIN = 2, max) %>% data.frame(max_soil_relabund = .) %>% 
  rownames_to_column("OTU") %>% arrange(desc(max_soil_relabund))

mean.soil.abunds.water.core <- OTUsREL[which(design$type == "soil"), high.activity.water.core$OTU] %>% 
  colMeans %>% data.frame(mean_soil_relabund = .) %>% 
  rownames_to_column("OTU") %>% arrange(desc(mean_soil_relabund))
max.soil.abunds.water.core <- OTUsREL[which(design$type == "soil"), high.activity.water.core$OTU] %>% 
  apply(X = ., MARGIN = 2, max) %>% data.frame(max_soil_relabund = .) %>% 
  rownames_to_column("OTU") %>% arrange(desc(max_soil_relabund))


soil.vs.lake.abunds <- high.activity.soil.core %>% 
  left_join(mean.soil.abunds.soil.core) %>% left_join(max.soil.abunds.soil.core) %>% 
  mutate(soil_is_source = ifelse(max_soil_relabund > 1e-3 & RNA.max > 1e-3, T, F)) %>% 
  mutate(Taxon = ifelse(Genus == "unclassified", paste(Family, "sp."), Genus))
```

## Figure 5: Soil vs. Lake Comparisons
```{r}
soil.vs.lake.abunds %>%   
  ggplot(aes(x = max_soil_relabund, y = RNA.max)) +
  geom_vline(xintercept = 1e-3, alpha = 0.1) +
  geom_hline(yintercept = 1e-3, alpha = 0.1) +
  geom_jitter(size = 3, aes(alpha = soil_is_source,color = soil_is_source), show.legend = F) +
  scale_x_log10() + 
  scale_y_log10() +
  scale_alpha_discrete(range = c(.3,.8)) +
  annotation_logticks(long = unit(.1, "in")) +
  scale_color_manual(values = my.cols) +
  labs(x = "Max Soil Relative Abundance", y = "Max Lake RNA \nRelative Abundance") +
  geom_text_repel(size = 5, data = subset(soil.vs.lake.abunds, soil_is_source == T), 
                   aes(label = Taxon), force = 1.5, alpha = 0.9, segment.alpha = 0.8, box.padding = .75, max.iter = 10000) +
  ggsave("../figures/05_soil-contribs-to-active-water-otus.pdf", bg = "white", height = 6, width = 7) +
  ggsave("../figures/05_soil-contribs-to-active-water-otus.png", dpi = 500, height = 6, width = 7)
```

# Ecosystem Functioning

## Fig 1: Microbial metabolism along reservoir gradient

Read in data 
```{r results='hide'}
metab <- read.table("../data/res.grad.metab.txt", sep="\t", header=TRUE)
colnames(metab) <- c("dist", "BP", "BR")
BGE <- round((metab$BP/(metab$BP + metab$BR)),3)
metab <- cbind(metab, BGE)


# Quadratic regression for BP
dist <- metab$dist
dist2 <- metab$dist^2
BP.fit <- lm(metab$BP ~ dist + dist2)
BP.R2 <- round(summary(BP.fit)$r.squared, 2)

# Simple linear regression for BR
BR.fit <- lm(metab$BR ~ metab$dist)
BR.R2 <- round(summary(BR.fit)$r.squared, 2)
BR.int <- BR.fit$coefficients[1]
BR.slp <- BR.fit$coefficients[2]

# Simple linear regression for BGE
BGE.fit <- lm(metab$BGE ~ metab$dist)
BGE.R2 <- round(summary(BGE.fit)$r.squared, 2)
BGE.int <- BGE.fit$coefficients[1]
BGE.slp <- BGE.fit$coefficients[2]

BP.R2
BR.R2
BGE.R2

BP.plot <- ggplot(metab, aes(x = dist, y = BP)) + 
  geom_point() + 
  geom_smooth(method = "lm", formula = y ~ x + I(x^2), color = "black") +
  annotate(geom = "text", x = 50, y = 1.5, size = 5, label = paste0("R^2== ",BP.R2), parse = T) +
  labs(y = expression(paste('BP (', mu ,'M C h'^-1* ')')), 
       x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m)"))) +
  scale_x_reverse(limits = c(400,0))
BR.plot <- ggplot(metab, aes(x = dist, y = BR)) + 
  geom_point() + 
  geom_smooth(method = "lm", formula = y ~ x, color = "black") + 
  annotate("text", x = 50, y = 1.5, size = 5, label = paste0("R^2== ",BR.R2), parse = T ) +
  labs(y = expression(paste('BR (', mu ,'M C h'^-1* ')')), 
       x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m)"))) +
  scale_x_reverse(limits = c(400,0))
BGE.plot <- ggplot(metab, aes(x = dist, y = BGE)) + 
  geom_point() + 
  geom_smooth(method = "lm", formula = y ~ x + I(x^2), color = "black") +
  annotate("text", x = 50, y = .5, size = 5, label = paste0("R^2== ",BGE.R2), parse = T ) +
  labs(y = "BGE", 
       x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m)"))) +
  scale_x_reverse(limits = c(400,0))
plot_grid(BP.plot + theme(axis.title.x = element_blank(), axis.text.x = element_blank(), 
                          plot.margin = unit(c(1, 1, -1, 0), "cm")), 
          BR.plot + theme(axis.title.x = element_blank(), axis.text.x = element_blank(),
                          plot.margin = unit(c(-1, 1, -1, 0), "cm")), 
          BGE.plot + theme(plot.margin = unit(c(-1, 1, 0, 0), "cm")), align = "hv", ncol = 1) + 
  ggsave("../figures/06_ecosystem-functions.pdf", bg = "white", width = 6, height = 11)
```

## Relation of ecosystem functions and community structure
```{r}
# detrend the spatial signal
bp.resid <- resid(lm(BP ~ dist + I(dist)^2, data = metab))
br.resid <- resid(lm(BR ~ dist, data = metab))


metab.resids <- metab
metab.resids$BR_resid <- br.resid + mean(metab$BR)
metab.resids$BP_resid <- bp.resid + mean(metab$BP)

transient.metabolism <- data.frame(transients = terr.REL, dist = design.dna$distance) %>% 
  left_join(metab.resids) 


bp.mod.quad <- lm(BP_resid ~ transients + I(transients^2), data = transient.metabolism)
bp.mod.lin <- lm(BP_resid ~ transients, data = transient.metabolism)
bp.mod.int <- lm(BP_resid ~ 1, data = transient.metabolism)
anova(bp.mod.int, bp.mod.lin, bp.mod.quad)
AIC(bp.mod.quad, bp.mod.lin, bp.mod.int)

br.mod.quad <- lm(BR_resid ~ transients + I(transients^2), data = transient.metabolism)
br.mod.lin <- lm(BR_resid ~ transients, data = transient.metabolism)
br.mod.int <- lm(BR_resid ~ 1, data = transient.metabolism)
anova(br.mod.int, br.mod.lin, br.mod.quad)
AIC(br.mod.int, br.mod.lin, br.mod.quad)

bge.mod.quad <- lm(BGE ~ transients + I(transients^2), data = transient.metabolism)
bge.mod.lin <- lm(BGE ~ transients, data = transient.metabolism)
bge.mod.int <- lm(BGE ~ 1, data = transient.metabolism)
anova(bge.mod.int, bge.mod.lin, bge.mod.quad)
AIC(bge.mod.int, bge.mod.lin, bge.mod.quad)

round(summary(br.mod.quad)$r.squared, 2)
round(summary(bp.mod.quad)$r.squared, 2)


total_core <- rowSums(OTUsREL[design$molecule == "DNA" & design$type == "water",
                               subset(rbind.data.frame(high.activity.water.core, 
                                                       high.activity.soil.core), RNA.max > .01)$OTU])

summary(lm(BP ~ transients * dist, transient.metabolism))
summary(lm(BR ~ transients * dist, transient.metabolism))


data.frame(
  soil_core = rowSums(OTUsREL[design$molecule == "DNA" & design$type == "water",
           subset(soil.vs.lake.abunds, RNA.max > .01)$OTU]), 
  dist = design.dna$distance) %>% 
  left_join(metab.resids) %>% select(-BGE, -BP, -BR) %>% gather(metab, value, -soil_core, -dist) %>% 
  ggplot(aes(x = soil_core, y = value, color = metab, fill = metab)) +
  geom_point(size = 2) + 
  geom_smooth(alpha = .25, method = 'lm', formula = y ~ x + I(x^2)) +
  labs(x = "Relative Abundance of Soil-derived Core",
       y = expression(paste('Metabolism (', mu ,'M C h'^-1* ')'))) +
  scale_color_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_fill_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  ggsave("../figures/06_soilcore-function.pdf", bg = "white", width = 7, height = 6)

data.frame(
  water_core = rowSums(OTUsREL[design$molecule == "DNA" & design$type == "water",
                               subset(high.activity.water.core, RNA.max > .01)$OTU]), 
  dist = design.dna$distance) %>% 
  left_join(metab.resids) %>% select(-BGE,-BR,-BP) %>% gather(metab, value, -water_core, -dist) %>% 
  ggplot(aes(x = water_core, y = value, color = metab, fill = metab)) +
  geom_point(size = 2) + 
  geom_smooth(alpha = .25, method = 'lm', formula = y ~ x + I(x^2)) +
  labs(x = "Relative Abundance of non-soil-derived Core",
       y = expression(paste('Metabolism (', mu ,'M C h'^-1* ')'))) +
  scale_color_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_fill_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  ggsave("../figures/06_nonsoilcore-function.pdf", bg = "white", width = 7, height = 6)



data.frame(transients = resid(lm(terr.REL ~ design.dna$distance)) + mean(terr.REL), dist = design.dna$distance) %>% 
  left_join(metab.resids) %>% select(-BGE, -BP, -BR) %>% gather(metab, value, -transients, -dist) %>% 
  ggplot(aes(x = transients, y = value, color = metab, fill = metab)) +
  geom_point(size = 2, show.legend = F) + 
  geom_smooth(alpha = .25, method = 'lm', formula = y ~ x, show.legend = F) +
  annotation_logticks(sides = "b") +
  labs(x = "Relative Abundance of Transient Taxa",
       y = expression(paste('Metabolism (', mu ,'M C h'^-1* ')'))) +
  scale_color_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_fill_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_y_continuous(limits = c(0,3)) +
  theme(plot.margin = unit(c(1,1,0,0), "cm")) +
  ggsave("../figures/06_transients-function.pdf", bg = "white", width = 7, height = 6)


core.metab <- data.frame(
  total_core = rowSums(OTUsREL[design$molecule == "DNA" & design$type == "water",
                               subset(rbind.data.frame(high.activity.water.core, 
                                                       high.activity.soil.core), RNA.max > .01)$OTU]), 
  dist = design.dna$distance) %>% 
  left_join(metab.resids)

summary(lm(BP ~ total_core * dist, core.metab))
summary(lm(BR ~ total_core + dist, core.metab))


core.metab <- data.frame(
  total_core = rowSums(OTUsREL[design$molecule == "DNA" & design$type == "water",
                               subset(rbind.data.frame(high.activity.water.core, 
                                                       high.activity.soil.core), RNA.max > .01)$OTU]), 
  dist = design.dna$distance) %>% 
  left_join(metab.resids)
core.metab$total_core_resid <- resid(lm(total_core ~ dist + I(dist^2), core.metab)) + mean(core.metab$total_core)
summary(lm(BP_resid ~ total_core, core.metab))
summary(lm(BR_resid ~ total_core + I(total_core^2), core.metab))


core.metab %>% select(-BGE, -BP, -BR, -total_core) %>% gather(metab, value, -total_core_resid, -dist) %>% 
  ggplot(aes(x = total_core_resid, y = value, color = metab, fill = metab)) +
  geom_point(size = 2, show.legend = F) + 
  geom_smooth(alpha = .25, method = 'lm', formula = y ~ x, show.legend = F) +
  labs(x = "Relative Abundance of Core Taxa",
       y = expression(paste('Metabolism (', mu ,'M C h'^-1* ')'))) +
  scale_color_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_fill_viridis("Ecosystem Function", discrete = T, begin = .1, end = .6, option = "D") +
  scale_y_continuous(limits = c(0,3)) +
  theme(plot.margin = unit(c(1,1,0,0), "cm")) +
  ggsave("../figures/06_core-function.pdf", bg = "white", width = 7, height = 6)



```

# Analyze environmental controls
```{r}
# library(AEM)
# # pull out the in-lake samples
# in.lake.dna.samples <- which(design$type == "water" & design$molecule == "DNA" & design$distance < 300)
# env <- env.dat[which(env.dat$sample.ID %in% design$station[in.lake.dna.samples]),c("temp", "pH", "DO", "TP", "chla")]
# env <- scale(env)
# geo_dist <- as.vector(dist(design[in.lake.dna.samples,]$distance))
# env_dist <- as.vector(dist(env))
# com_dist <- as.vector(vegdist(decostand(lake.tot, "total")))
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = env_dist, y = com_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = geo_dist, y = com_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = env_dist, y = geo_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# 
# # construct asymetric eigenvector maps along transect
# n <- nrow(env)
# aem.out <- aem.time(n, moran = T, plot.moran = T)
# colnames(aem.out$aem) <- paste0("AEM", seq(1, n-1))
# aems <- aem.out$aem[,which(aem.out$Moran$p.value < 0.05)]
# 
# 
# prcomp(env, scale. = T)
# round(cor(cbind(aems, env)),2)
# lake.tot <- OTUs[in.lake.dna.samples,]
# 
# vp.tot.pos <- varpart(lake.tot, aems[,c("AEM1", "AEM2", "AEM3")], env, transfo = "hellinger")
# vp.tot.pos
# plot(vp.tot.pos)
# 
# vp.tot.neg <- varpart(lake.tot, aems[,c("AEM8", "AEM9", "AEM10")], env, transfo = "hellinger")
# vp.tot.neg
# plot(vp.tot.neg)
# 
# # rna
# in.lake.rna.samples <- which(design$type == "water" & design$molecule == "RNA" & design$distance < 300)
# env <- env.dat[which(env.dat$sample.ID %in% design$station[in.lake.rna.samples]),c("temp", "pH", "DO", "TP", "chla")]
# env <- scale(env)
# 
# # construct asymetric eigenvector maps along transect
# n <- nrow(env)
# aem.out <- AEM::aem.time(n, moran = T, plot.moran = T)
# colnames(aem.out$aem) <- paste0("AEM", seq(1, n-1))
# aems <- aem.out$aem[,which(aem.out$Moran$p.value < 0.05)]
# 
# prcomp(env, scale. = T)
# round(cor(cbind(aems, env)),2)
# lake.act <- OTUs[in.lake.rna.samples,]
# 
# geo_dist <- as.vector(dist(design[in.lake.rna.samples,]$distance))
# env_dist <- as.vector(dist(env))
# com_dist <- as.vector(vegdist(decostand(lake.act, "total")))
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = env_dist, y = com_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = geo_dist, y = com_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# data.frame(env_dist = resid(lm(env_dist ~ geo_dist)), 
#            com_dist = com_dist,
#            geo_dist = resid(lm(geo_dist ~ env_dist))) %>% 
#   ggplot(aes(x = env_dist, y = geo_dist)) + 
#   geom_point(alpha = 0.5) + 
#   geom_smooth(method = 'lm')
# 
# vp.act.pos <- varpart(lake.act, aems[,c("AEM1", "AEM2", "AEM3")], env, transfo = "hellinger")
# vp.act.pos
# plot(vp.act.pos)
# 
# vp.act.neg <- varpart(lake.act, aems[,c("AEM8", "AEM9", "AEM10")], env, transfo = "hellinger")
# vp.act.neg
# plot(vp.act.neg)
```

# null model to control for richness
```{r}
# require("progress")
# 
# Calculate abundance-weighted Raup-Crick dissimilarities
# rcbraycurtis <- function(OTUs){
#   regional.abunds <- t(as.matrix(colSums(OTUs)))
#   regional.relabunds <- decostand(regional.abunds, method = "total")
#   occupancy.probs <- t(as.matrix(colSums(decostand(OTUs, method = "pa")) / nrow(OTUs)))
#   site.abunds <- rowSums(OTUs)
#   site.rich <- specnumber(OTUs)
#   a <- regional.relabunds * occupancy.probs
# 
#   # Create a null community based on Stegen et al. 2015
#   r <- nrow(OTUs)
#   c <- ncol(OTUs)
#   spec.vec <- 1:ncol(OTUs)
#   RCbc.nulls <- array(NA, c(r, r, 999))
# 
#   # stochastic community assembly nulls
#   for(i in 1:999){
#     if(i == 1) pb <- progress_bar$new(total = 999, force = T)
#     pb$update(ratio = i/999)
# 
#     null.comm <- OTUs * 0
#     # for first simulation:
#     for(row.i in 1:nrow(null.comm)){
#       #print(paste("run :", i, " -> ", row.i, " : ", site.abunds[row.i], " inds"))
# 
#       while(rowSums(null.comm)[row.i] < site.abunds[row.i]){
# 
# 
#         # choose a species based on its occupancy
#         local.specs <- sample(x = spec.vec, size = site.rich[row.i],
#                               prob = as.vector(occupancy.probs), replace = FALSE)
# 
#         local.probs <- decostand(t(as.matrix(regional.abunds[,local.specs])), method = "total")
# 
#         local.inds <- sample(x = local.specs, size = site.abunds[row.i],
#                              prob = as.vector(local.probs), replace = TRUE)
# 
#         local.abunds <- rle(sort(local.inds))
# 
#         # add an individual to the local community
#         null.comm[row.i, local.abunds$values] <- local.abunds$lengths
#       }
#     }
#     null.bc <- as.matrix(vegdist(decostand(null.comm, method = "log"), method = "bray"))
#     RCbc.nulls[,,i] <- null.bc
#   }
# }
# 
# RCbc.nulls.total <- rcbraycurtis(OTUs = OTUs[which(design$molecule == "DNA"),])
# RCbc.nulls.active <- rcbraycurtis(OTUs = OTUs[which(design$molecule == "RNA"),])
# 
# saveRDS(RCbc.nulls.total, file = "intermediate-data/RCbc.null.total.rda")
# saveRDS(RCbc.nulls.active, file = "intermediate-data/RCbc.null.active.rda")
# 
# RCbc.nulls.total <- readRDS(file = "intermediate-data/RCbc.null.total.rda")
# RCbc.nulls.active <- readRDS(file = "intermediate-data/RCbc.null.active.rda")
# 
# obs.bc.total <- as.matrix(vegdist(OTUsREL[which(design$molecule == "DNA"),], method = "bray"))
# obs.bc.active <- as.matrix(vegdist(OTUsREL[which(design$molecule == "RNA"),], method = "bray"))
# 
# get.rcbrayvals <- function(obs.bc, RCbc.nulls){
#   r <- nrow(as.matrix(obs.bc))
#   site.compares <- expand.grid(site1 = 1:r, site2 = 1:r)
#   RC.bray <- matrix(NA, nrow = r, ncol = r)
# 
#   for(row.i in 1:nrow(site.compares)){
#     site1 <- site.compares[row.i,1]
#     site2 <- site.compares[row.i,2]
#     pairwise.null <- RCbc.nulls[site1,site2,]
#     pairwise.bray <- obs.bc[site1,site2]
#     num.greater <- sum(pairwise.null > pairwise.bray)
#     num.ties <- sum(pairwise.null == pairwise.bray)
#     val <- (((1 * num.greater) + (0.5 * num.ties))/1000 - 0.5) * 2
#     RC.bray[site1, site2] <- val
#   }
#   return(RC.bray)
# }
# 
# RC.bray.active <- get.rcbrayvals(obs.bc.active, RCbc.nulls.active)
# rownames(RC.bray.active) <- rownames(subset(design, molecule == "RNA"))
# colnames(RC.bray.active) <- rownames(subset(design, molecule == "RNA"))
# RC.bray.active.dist <- as.dist(RC.bray.active)
# 
# RC.bray.total <- get.rcbrayvals(obs.bc.total, RCbc.nulls.total)
# rownames(RC.bray.total) <- rownames(subset(design, molecule == "DNA"))
# colnames(RC.bray.total) <- rownames(subset(design, molecule == "DNA"))
# RC.bray.total.dist <- as.dist(RC.bray.total)
# 
# # Similarity to Soil Sample
# UL.bray      <- 1-as.matrix(vegdist(OTUsREL.log, method="bray"))
# UL.bray.lake <- UL.bray[-c(1:3), 1:3] 
# bray.mean    <- round(apply(UL.bray.lake, 1, mean), 3)
# bray.se      <- round(apply(UL.bray.lake, 1, se), 3)
# UL.sim       <- cbind(design[-c(1:3), ], bray.mean, bray.se)
# 
# 
# # Calculate Linear Model
# model.terr <- lm(UL.sim$bray.mean ~ UL.sim$distance * UL.sim$molecule)
# 
# # Calculate Confidance Intervals of Model
# newdata.terr <- data.frame(cbind(UL.sim$molecule, UL.sim$distance))
# conf95.terr <- predict(model.terr, newdata.terr, interval="confidence")
# 
# # Dummy Variables Regression Model ("Terrestrial Influence")
# D2 <- (UL.sim$molecule == "RNA")*1
# fit.Fig.3b <- lm(UL.sim$bray.mean ~ UL.sim$distance + D2 + UL.sim$distance*D2)
# D2.R2 <- round(summary(fit.Fig.3b)$r.squared, 2)
# summary(fit.Fig.3b)
# 
# DNA.int.3b <- fit.Fig.3b$coefficients[1]
# DNA.slp.3b <- fit.Fig.3b$coefficients[2]
# RNA.int.3b <- DNA.int.3b + fit.Fig.3b$coefficients[3]
# RNA.slp.3b <- DNA.slp.3b + fit.Fig.3b$coefficients[4]
# 
# 
# UL.sim %>% 
#   mutate(molecule = ifelse(UL.sim$molecule == "DNA", "Total", "Active")) %>% 
#   ggplot(aes(x = distance, y = bray.mean, color = molecule, fill = molecule)) +
#   geom_point() + 
#   geom_smooth(method = "lm") + 
#   labs(y = "Community Similarity to Soils", 
#        x = (expression("Soil" %->% "Riverine" %->% "Lacustrine" %->% "Dam (m)"))) + 
#   scale_color_manual("Community Subset", values = my.cols) + 
#   scale_fill_manual("Community Subset", values = my.cols) +
#   scale_x_reverse(limits = c(400,0)) + 
#   annotate(geom = "text", x = 50, y = 0.2, size = 5,
#            label = paste0("R^2== ",round(summary(fit.Fig.3b)$r.squared, 2)), parse = T) +
#   ggsave("../figures/03_similarity-to-soils.pdf", bg = "white", height = 6, width = 8)
# 
# RC.bray.total.dist
```

## Import Phototrophs
```{r}
# # The phototrophs
# cyanos.in <- "../data/UL.cyano.final.shared"
# phytos.in <- "../data/UL.euks.final.shared"
# 
# cyanos <- read.otu(shared = cyanos.in, cutoff = "0.03")
# phytos <- read.otu(shared = phytos.in, cutoff = "0.03")
# 
# # Remove OTUs with less than two occurences across all sites
# cyanos <- cyanos[, which(colSums(cyanos) >= 2)]
# phytos <- phytos[, which(colSums(phytos) >= 2)]
# 
# # Remove sites where we have low coverage
# cyanos <- cyanos[-which(coverage < 10000), ]
# phytos <- phytos[-which(coverage < 10000), ]
# 
# # Remove Non Intersecting Sites
# ratio.sites <- intersect(intersect(rownames(cyanos), rownames(phytos)), rownames(OTUs))
# cyanos <- cyanos[ratio.sites, ]
# phytos <- phytos[ratio.sites, ]
# heteros <- OTUs[ratio.sites, ]
# design.int <- design[ratio.sites, ]
# 
# # Remove RNA Sites
# DNA.samps <- which(design.int$molecule == "DNA")
# cyanos <- cyanos[DNA.samps, ]
# phytos <- phytos[DNA.samps, ]
# heteros <- OTUs[DNA.samps, ]
# design.dna <- design[DNA.samps, ]
# 
# # Observed Richness
# S.cyano <- rowSums((cyanos > 0) * 1)
# S.phyto <- rowSums((phytos > 0) * 1)
# S.hetero <- rowSums((heteros > 0) * 1)
# 
# N.cyano <- rowSums(cyanos)
# N.phyto <- rowSums(phytos)
# N.hetero <- rowSums(heteros) - rowSums(cyanos)
#                     
# HtoC <- N.hetero / N.cyano
# HtoP <- N.hetero / N.phyto
# HtoBoth <- N.hetero / (N.cyano + N.phyto)
# 
# data_frame(S.cyano, S.phyto, S.hetero, 
#            metric = "richness", 
#            location = design.dna$distance) %>% 
#   gather(group, richness, -metric, -location) %>% 
#   filter(!is.na(location)) %>% 
#   ggplot(aes(x = location, y = richness, color = group)) +
#   geom_jitter(alpha = 0.5) +
#   geom_smooth() +
#   scale_y_log10() +
#   scale_x_reverse()
# 
# data_frame(N.cyano, N.phyto, N.hetero, 
#            metric = "abundance", 
#            location = design.dna$distance) %>% 
#   gather(group, abundance, -metric, -location) %>% 
#   filter(!is.na(location)) %>% 
#   ggplot(aes(x = location, y = abundance, color = group)) +
#   geom_jitter(alpha = 0.5) +
#   geom_smooth() +
#   scale_y_log10() +
#   scale_x_reverse()
# 
# data_frame(HtoC, HtoP, HtoBoth, 
#            metric = "ratio", 
#            location = design.dna$distance) %>% 
#   gather(group, ratio, -metric, -location) %>% 
#   filter(!is.na(location)) %>% 
#   ggplot(aes(x = location, y = ratio, color = group)) +
#   geom_jitter(alpha = 0.5) +
#   geom_smooth() +
#   scale_y_log10() +
#   scale_x_reverse()
```

# Old Figures
## Figure 2: Bacterial Richness and Terrestrial Influence Across Gradient
```{r}
# # Set Plot Symbol Parameters
# mol <- rep(NA, length(lake$molecule))
#   for (i in 1:length(mol)){
#     if (lake$molecule[i] == "DNA"){
#       mol[i] <- 22
#     } else {
#       mol[i] <- 24
#     }
#   }
# cols <- rep(NA, length(lake$molecule))
#   for (i in 1:length(cols)){
#     if (lake$molecule[i] == "DNA"){
#       cols[i] <- "gray15"
#     } else {
#       cols[i] <- "gray75"
#     }
#   }
# 
# # Initial Plot
# png(filename="../figures/Figure2.png",
#     width = 1200, height =1200, res = 96*2)
# 
# par(mfrow = c(1,1), mar = c(0, 5, 0, 1) + 0.5, oma = c(4, 2, 0, 0) + 0.5)
# bar.layout <- layout(rbind(1, 2), height = c(4, 4)) 
# 
# # Richness Across Gradient Plot
# plot(lake$S.obs ~ lake$distance, col= "black", bg = cols, pch=mol, las = 1,
#      xlim = c(400, -15), ylim = c(0, 2750), cex = 1.5, 
#      xlab="", ylab="", xaxt="n")
# 
# #   matlines(lake$distance[lake$molecule == "DNA"], conf95.rich[lake$molecule == "DNA", ],
# #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# #   matlines(lake$distance[lake$molecule == "RNA"], conf95.rich[lake$molecule == "RNA", ],
# #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# 
# # Add multiple regression lines
# clip(400, 0, 0, 2800)
# abline(a = DNA.int.3a, b = DNA.slp.3a, col = "black", lwd = 2.5, lty = 6)
# abline(a = RNA.int.3a, b = RNA.slp.3a, col = "black", lwd = 2.5, lty = 4)
# text(40, 1500, labels = bquote(italic(R)^2 == .(D1.R2)), cex = 1)
# 
# axis(side = 1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, las = 1,
#      at = c(0, 100, 200, 300, 400))
# axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))
# axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400)) 
# axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))     
# axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# 
# mtext("Richness\n(S)" , side = 2, line = 4, cex=1.5)
# 
# legend("topright", legend = levels(lake$molecule), pch=c(22, 24), 
#        pt.bg = c("gray15", "gray75"), bty='n', cex = 1, ncol=2)
# 
# box(lwd=2)
# 
# # Terrestrial Influence Plot
# 
# plot(UL.sim$bray.mean ~ UL.sim$distance, col= "black", bg = cols, pch=mol, 
#      las = 1, xlim = c(400, -15), ylim = c(0, 0.25), cex = 1.5, 
#      xlab="", ylab="", xaxt="n")
# 
# #   matlines(lake$distance[lake$molecule == "DNA"], conf95.terr[lake$molecule == "DNA", ],
# #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# #   matlines(lake$distance[lake$molecule == "RNA"], conf95.terr[lake$molecule == "RNA", ],
# #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# 
# # Add multiple regression lines
# clip(400, 0, 0, 0.27)
# abline(a = DNA.int.3b, b = DNA.slp.3b, col = "black", lwd = 2.5, lty = 6)
# abline(a = RNA.int.3b, b = RNA.slp.3b, col = "black", lwd = 2.5, lty = 4)
# text(40, 0.125, labels = bquote(italic(R)^2 == .(D2.R2)), cex = 1)
# 
# axis(side = 1, lwd.ticks = 2, cex.axis = 1, las = 1,
#      labels = c("400", "300", "200", "100", "0"), at = c(400, 300, 200, 100, 000))
# axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=-0.02, labels = F,  cex.axis = 2, las = 1, 
#      at = c(400, 300, 200, 100, 000))
# axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(400, 300, 200, 100, 000))
# axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(400, 300, 200, 100, 000))
# axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# 
# mtext("Terrestrial\nInfluence" , side = 2, line = 4, cex=1.5)
# mtext("Distance (m)" , side = 1, line = 3, cex=1.5)
# 
# # legend("topright", legend = levels(UL.sim$molecule), pch=c(22, 24), 
# #        pt.bg = c("gray15", "gray75"), bty='n', cex = 1.25)
# 
# box(lwd=2)
# 
# # # Lake Influence Plot
# # # plot(UL.sim2$DNA ~ UL.sim2$distance, col= "black", bg = cols, pch=mol, las = 1,
# # #      xlim = c(400, 0), ylim = c(0, 1), cex = 1.5, 
# # #      xlab="", ylab="")
# # 
# # #   matlines(lake$distance[lake$molecule == "DNA"], conf95.lake[lake$molecule == "DNA", ],
# # #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# # #   matlines(lake$distance[lake$molecule == "RNA"], conf95.lake[lake$molecule == "RNA", ],
# # #          lty = c(1, 0, 0), col=c("black", "gray50", "gray50"), lwd=c(2, 1, 1))
# # 
# # # Add multiple regression lines
# # clip(400, 0, 0, 1)
# # abline(a = DNA.int.3c, b = DNA.slp.3c, col = "black", lwd = 2.5, lty = 6)
# # 
# # clip(400, 0, 0, 1)
# # abline(a = RNA.int.3c, b = RNA.slp.3c, col = "black", lwd = 2.5, lty = 4)
# # 
# # axis(side = 1, lwd.ticks = 2, cex.axis = 1, las = 1)
# # axis(side = 2, lwd.ticks = 2, cex.axis = 1, las = 1)
# # axis(side = 3, lwd.ticks = 2, tck=-0.05, labels = F,  cex.axis = 2, las = 1)  
# # axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# # axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)         
# # axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# # axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)   
# # axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# # 
# # mtext("Distance (m)" , side = 1, line = 3, cex=1.5)
# # mtext("Lake\nInfluence", side = 2, line = 4, cex=1.5)
# # 
# # legend("topleft", legend = levels(UL.sim$molecule), pch=c(22, 24), 
# #        pt.bg = c("gray15", "gray75"), bty='n', cex = 1.25)
# # 
# # box(lwd=2)
# 
# # Close Plot Defice
# dev.off()
# graphics.off() 
# par(opar)
```
## Figure 3: Soil Organisms Plot
```{r}
# # Initial Plot
# png(filename="../figures/Figure3.png",
#     width = 1200, height =1200, res = 96*2)
# 
# par(mfrow = c(1,1), mar = c(0, 7, 0, 1) + 0.5, oma = c(4, 2, 0, 0) + 0.5)
# bar.layout <- layout(rbind(1, 2), height = c(4, 4)) 
# 
# # Soil OTU Richness Across Gradient Plot
# plot(terr.rich.log ~ design.dna$distance, col= "black", pch=22, las = 1,
#      xlim = c(400, -15), ylim = c(1.5, 3.5), cex = 1.5,  
#      xlab="", ylab="", xaxt="n", yaxt="n")
# 
# clip(0, 375, 1.5, 3.4)
# abline(a = T1.int, b = T1.slp, col = "black", lwd = 2.5, lty = 6)
# text(40, 3, labels = bquote(italic(R)^2 == .(T1.R2)), cex = 1)
# 
# axis(side = 1, lwd.ticks = 2, tck = -0.02, labels = F, cex.axis = 1, las = 1)
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1) 
# axis(side = 3, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 2, lwd.ticks = 2, at = c(2, 3), labels = c(10^2, 10^3), cex.axis = 1, las = 1)
# axis(side = 2, lwd.ticks = 2, tck = -0.02, at = log10(c(seq(10, 100, by = 10), 
#      seq(100, 1000, by = 100), seq(1000, 10000, by = 1000))), labels = F, cex.axis = 1, las = 1)
# axis(side = 2, lwd.ticks = 2, at = c(2, 3), tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 2, lwd.ticks = 2, tck = 0.005, at = log10(c(seq(10, 100, by = 10), 
#      seq(100, 1000, by = 100), seq(1000, 10000, by = 1000))), labels = F, cex.axis = 1, las = 1)    
# axis(side = 4, lwd.ticks = 2, at = c(2, 3), tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 4, lwd.ticks = 2, at = c(2, 3), tck=0.02, labels = F,  cex.axis = 2, las = 1)
# axis(side = 4, lwd.ticks = 2, tck = 0.01, at = log10(c(seq(10, 100, by = 10), 
#      seq(100, 1000, by = 100), seq(1000, 10000, by = 1000))), labels = F, cex.axis = 1, las = 1)
# 
# mtext("Transient\nRichness\n(S)" , side = 2, line = 4, cex=1.5)
# box(lwd=2)
# 
# # Soil OTU Relative Abundance Across Gradient Plot
# plot(terr.REL.log ~ design.dna$distance, col= "black", pch=22, las = 1,
#      xlim = c(400, -15), ylim = c(-2.5, -.5), cex = 1.5, 
#      xlab="", ylab="", xaxt="n", yaxt="n")
# 
# clip(0, 375, -2.5, -0.5)
# abline(a = T2.int, b = T2.slp, col = "black", lwd = 2.5, lty = 6)
# text(40, -1, labels = bquote(italic(R)^2 == .(T2.R2)), cex = 1)
# 
#   axis(side = 1, lwd.ticks = 2, labels = T, cex.axis = 1, las = 1)
#   axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1) 
#   axis(side = 3, lwd.ticks = 2, tck=-0.02, labels = F,  cex.axis = 2, las = 1)
#   axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
#   axis(side = 2, lwd.ticks = 2, at = c(-2, -1), labels = c(0.01, 0.1), cex.axis = 1, las = 1)
#   axis(side = 2, lwd.ticks = 2, tck = -0.02, at = log10(c(seq(0.001, 0.01, by = 0.001), 
#        seq(0.01, 0.1, by = 0.01), seq(0.1, 1, by = 0.1))), labels = F, cex.axis = 1, las = 1)
#   axis(side = 2, lwd.ticks = 2, at = c(-2, -1), tck=0.01, labels = F,  cex.axis = 2, las = 1)
#   axis(side = 2, lwd.ticks = 2, tck = 0.005, at = log10(c(seq(0.001, 0.01, by = 0.001), 
#        seq(0.01, 0.1, by = 0.01), seq(0.1, 1, by = 0.1))), labels = F, cex.axis = 1, las = 1)    
#   axis(side = 4, lwd.ticks = 2, at = c(-2, -1), tck=-0.01, labels = F,  cex.axis = 2, las = 1)
#   axis(side = 4, lwd.ticks = 2, at = c(-2, -1), tck=0.02, labels = F,  cex.axis = 2, las = 1)
#   axis(side = 4, lwd.ticks = 2, tck = 0.01, at = log10(c(seq(0.001, 0.01, by = 0.001), 
#        seq(0.01, 0.1, by = 0.01), seq(0.1, 1, by = 0.1))), labels = F, cex.axis = 1, las = 1)
# 
# mtext("Distance (m)" , side = 1, line = 3, cex=1.5)
# mtext("Transient\nRelative\nAbundance", side = 2, line = 4, cex=1.5)
# 
# box(lwd=2)  
#   
# 
# # Close Plot Defice
# dev.off()
# graphics.off() 
```

## Core
## Model Fit
```{r}
# 
# terrestrial <- data.frame(in.lake.core.soil.REL, design.dna$distance)
# terrestrial$distance[1] <- 10^-8
# colnames(terrestrial) <- c("t.rel.abund", "distance")
# 
# mm <- function(x,V,K){
#   (V * x)/(K + x)
# }
# 
# fit.t.1 <- lm(terrestrial$t.rel.abund ~ terrestrial$distance)
# fit.t.2 <- lm(terrestrial$t.rel.abund ~ poly(terrestrial$distance,2,raw=TRUE))
# fit.t.3 <- lm(terrestrial$t.rel.abund ~ poly(terrestrial$distance,3,raw=TRUE))
# fit.t.mm <- nls(t.rel.abund ~ mm((400-distance[1:15]), max, halfsat), data = terrestrial[1:15,], start = list(max = 0.75, halfsat = (400-325)), trace = T)
# summary(fit.t.mm)$coefficients
# 
# lake <- data.frame(in.lake.core.water.REL, design.dna$distance)
# lake$distance[1] <- 10^-8
# colnames(lake) <- c("l.rel.abund", "distance")
# 
# fit.l.1 <- lm(lake$l.rel.abund ~ lake$distance)
# fit.l.2 <- lm(lake$l.rel.abund ~ poly(lake$distance,2,raw=TRUE))
# fit.l.3 <- lm(lake$l.rel.abund ~ poly(lake$distance,3,raw=TRUE))
# fit.l.mm <- nls(l.rel.abund ~ mm((400-distance), max, halfsat), data = lake[1:15,], start = list(max = 0.20, halfsat = (400-300)), trace = T)
# #fit.l.pow = nls(l.rel.abund ~ I(distance^power), data=lake, start=list(power = -0.06590247), trace = T)
```

## Figure 4: Plot Core Community
```{r}
# png(filename="../figures/Figure4.png",
#     width = 1000, height = 1000, res = 96*2)
# par(mar = c(5,7,4,1))
# plot(in.lake.core.soil.REL ~ design.dna$distance, 
#      ylim = c(0, 1), xlim = c(400, -15), pch = 22, bg = "black", 
#      ylab = "", xlab = "", xaxt = "n", yaxt = "n", cex = 2, cex.lab = 2)
# points(in.lake.core.water.REL ~ design.dna$distance, cex = 2, pch = 22)
# mtext("Distance (m)", side = 1, line = 3, cex = 1.5)
# mtext("Relative abundance of \ncore microbiome", side = 2, line = 3, cex = 1.5)
# axis(side = 1, lwd.ticks = 2, labels = T, cex.axis = 1, las = 1)
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1) 
# axis(side = 3, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 2, lwd.ticks = 2, at = c(0, .25, .5, .75, 1), cex.axis = 1, las = 1)
# axis(side = 2, lwd.ticks = 2, at = c(0, .25, .5, .75, 1), tck=0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 4, lwd.ticks = 2, at = c(0, .25, .5, .75, 1), tck=-0.01, labels = F,  cex.axis = 2, las = 1)
# axis(side = 4, lwd.ticks = 2, at = c(0, .25, .5, .75, 1), tck=0.01, labels = F,  cex.axis = 2, las = 1)
# 
# legend("topright", c("Terrestrial", "Lake"), col = c("black", "black"),
#          pt.bg = c("black", "white"), pch = 22, cex = 1.25, bty = "n")
# 
# box(lwd=2)
# 
# x <- lake$distance[1:15]
# 
# #lines(terrestrial$distance, predict(fit.t.2, data.frame(x=x)))
# 
# #lines(lake$distance, predict(fit.l.2, data.frame(x=x)))
# 
# #power <- round(summary(fit.t.pow)$coefficients[1], 3)
# #power.se <- round(summary(m)$coefficients[2], 3)
# 
# lines(x, predict(fit.t.mm, list(x = x)), lwd = 2.5, lty = 6)
# lines(x, predict(fit.l.mm, list(x = x)), lwd = 2.5, lty = 6)
# 
# dev.off()
# graphics.off() 
```

## Figure S2: chemical and physical variables along reservoir gradient
```{r }
# 
# # Start Plotting File
# png(filename="../figures/FigureS2.png",
#     width = 1500, height = 1200, res = 96*2)
# 
# par(mfrow = c(2,2))
# par(mar = c(5, 6, 1, 2) + 0.5)
# 
# # Total Phosphorus
# TP <- plot(rev(env.dat$dist.dam), env.dat$TP, 
#      ylab = "", xlab = "", cex.lab = 2, las = 1,
#      ylim = c(0,140), xlim = c(-15, 400),
#      pch = 22, cex = 2, bg = "white", col = "black", lwd = 2,
#      yaxt = "n", xaxt = "n")
# box(lwd = 2)
# 
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("0", "40", "80", "120"), at = c(0, 40, 80, 120))
#                               
# axis(side = 1, lwd.ticks = 2,  labels = F, cex.axis = 2, las = 1, mgp = c(3, 1.5, 0),
#    #labels = c("0", "100", "200", "300", "400"), 
#    at = c(0, 100, 200, 300, 400))
# 
# 
# mtext(expression(paste('Total Phosphorus (',mu,'g P L'^-1*')')), side = 2, line = 4, cex = 1)
# 
# par(mar = c(5, 5, 1, 3) + 0.5)
# 
# # Chlorophyll
# chla <- plot(rev(env.dat$dist.dam), env.dat$chla, 
#      ylab = "", xlab = "", cex.lab = 2, las = 1,
#      ylim = c(0,30), xlim = c(-15, 400),
#      pch = 22, cex = 2, bg = "white", col = "black", lwd = 2,
#      yaxt = "n", xaxt = "n")
# box(lwd = 2)
# 
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("0", "10", "20", "30"), at = c(0, 10, 20, 30))
# 
# axis(side = 1, lwd.ticks = 2, labels = F, cex.axis = 2, las = 1, mgp = c(3, 1.5, 0),
#    #labels = c("0", "100", "200", "300", "400"), 
#    at = c(0, 100, 200, 300, 400))
# 
# mtext(expression(paste('Chlorophyll a (',mu,'g L'^-1*')')), side = 2, line = 4, cex = 1)
# 
# par(mar = c(5, 6, 0, 2) + 0.5)
# #Dissolved Oxygen
# plot(rev(env.dat$dist.dam), env.dat$DO, 
#      ylab = "", xlab = "", cex.lab = 2, las = 1,
#      ylim = c(5,10), xlim = c(-15, 400),
#      pch = 22, cex = 2, bg = "white", col = "black", lwd = 2,
#      yaxt = "n", xaxt = "n")
# box(lwd = 2)
# 
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("5", "7.5", "10"), at = c(5,7.5, 10))
# 
# axis(side = 1, lwd.ticks = 2, cex.axis = 1.5, las = 1, mgp = c(3, 1.5, 0),
#    labels = c("0", "100", "200", "300", "400"), 
#    at = rev(c(0, 100, 200, 300, 400)))
# 
# mtext(expression(paste('Dissolved Oxygen (mg L'^-1*')')), side = 2, line = 4, cex = 1)
# 
# text(x = 35, y = 5.1, "STREAM", font = 2)
# text(x = 375, y = 5.1, "DAM", font = 2)
# 
# 
# #pH
# par(mar = c(5, 5, 0, 3) + 0.5)
# plot(rev(env.dat$dist.dam), env.dat$pH, 
#      ylab = "", xlab = "", cex.lab = 2, las = 1,
#      ylim = c(8,9), xlim = c(-15, 400),
#      pch = 22, cex = 2, bg = "white", col = "black", lwd = 2,
#      yaxt = "n", xaxt = "n")
# box(lwd = 2)
# 
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("8", "8.5","9"), at = c(8, 8.5, 9))
# 
# axis(side = 1, lwd.ticks = 2, cex.axis = 1.5, las = 1, mgp = c(3, 1.5, 0),
#    labels = c("0", "100", "200", "300", "400"), 
#    at = rev(c(0, 100, 200, 300, 400)))
# 
# mtext("pH", side = 2, line = 4, cex = 1)
# 
# text(x = 35, y = 8.02, "STREAM", font = 2)
# text(x = 375, y = 8.02, "DAM", font = 2)
# 
# dev.off() # this writes plot to folder
# graphics.off() # shuts down open devices
```

# Figure 1: Microbial Processes Across the Gradient
```{r results='hide'}
# png(filename="../figures/Figure1.png",
#     width = 1200, height =1200, res = 96*2)
# par(mfrow = c(1,1), mar = c(0, 5, 0, 1) + 0.5, oma = c(6, 2, 0, 0) + 0.5)
# bar.layout <- layout(rbind(1, 2, 3), height = c(3, 3, 3)) 
# #layout.show(bar.layout)
# 
# # Baterial Producivity (BP)
# plot(metab$dist, metab$BP, ylab = "", xlab = "", pch = 22, ylim = c(0, 2), 
#      xlim = c(400, -15), cex = 2, bg = "white", col = "black", cex.lab = 2, 
#      las = 1, lwd = 2, yaxt = "n", xaxt = "n")
# 
# axis(side = 1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, las = 1, 
#      at = c(0, 100, 200, 300, 400))
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("0.0", "1.0", "2.0"), at = c(0, 1.0, 2.0))
# axis(side = 3, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))
# axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 1.0, 2.0))
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#      at = c(0, 100, 200, 300, 400)) 
# axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#     at = c(0, 1.0, 2.0))
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))     
# axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 1.0, 2.0))
# 
# box(lwd = 2)
# 
# mtext(expression(paste('BP (', mu ,'M C h'^-1* ')')), side = 2, line = 4, cex = 1.25)
# 
# # Quadratic regression for BP
# dist <- metab$dist
# dist2 <- metab$dist^2
# BP.fit <- lm(metab$BP ~ dist + dist2)
# BP.R2 <- round(summary(BP.fit)$r.squared, 2)
# dist.vals <- seq(0, 375, 25)
# BP.pred <- predict(BP.fit,list(dist = dist.vals, dist2 = dist.vals^2))
# lines(dist.vals, BP.pred, col = "black", lwd = 2.5, lty = 6)
# text(40, 1.8, labels = bquote(italic(R)^2 == .(BP.R2)), cex = 1.5)
# 
# # Bacterial Respiration (BR)
# plot(metab$dist, metab$BR, ylab = "", xlab = "", pch = 22, ylim = c(0, 4), 
#      xlim = c(400, -15), cex = 2, bg = "white", col = "black", cex.lab = 2, 
#      las = 1, lwd = 2, yaxt = "n", xaxt = "n")
# 
# axis(side = 1, lwd.ticks = 2, tck=-0.02, labels = F, cex.axis = 1, las = 1, 
#      at = c(0, 100, 200, 300, 400))
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("0.0", "2.0", "4.0"), at = c(0, 2, 4))
# axis(side = 3, lwd.ticks = 2, tck=-0.02, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))
# axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 2, 4))
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#      at = c(0, 100, 200, 300, 400)) 
# axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#     at = c(0, 2, 4))
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))     
# axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 2, 4))
# 
# box(lwd = 2)
# 
# mtext(expression(paste('BR (', mu ,'M C h'^-1* ')')), side = 2, line = 4, cex = 1.25)
# 
# # Simple linear regression for BR
# BR.fit <- lm(metab$BR ~ metab$dist)
# BR.R2 <- round(summary(BR.fit)$r.squared, 2)
# BR.int <- BR.fit$coefficients[1]
# BR.slp <- BR.fit$coefficients[2]
# clip(0, 375, 0, 4.1)
# abline(a = BR.int, b = BR.slp, col = "black", lwd = 2.5, lty = 6)
# text(40, 3.75, labels = bquote(italic(R)^2 == .(BR.R2)), cex = 1.5)
# 
# # Bacterial Growth Efficiency
# plot(metab$dist, metab$BGE, ylab = "", xlab = "", pch = 22, ylim = c(0, 0.6), 
#      xlim = c(400, -15), cex = 2, bg = "white", col = "black", cex.lab = 2, 
#      las = 1, lwd = 2, yaxt = "n", xaxt = "n")
# 
# axis(side = 1, lwd.ticks = 2, cex.axis = 1.5, las = 1, 
#      labels = c("400", "300", "200", "100", "0"), at = c(400, 300, 200, 100, 000))
# axis(side = 2, lwd.ticks = 2, cex.axis = 1.5, las = 1,
#     labels = c("0.0", "0.3", "0.6"), at = c(0, 0.3, 0.6))
# axis(side = 3, lwd.ticks = 2, tck=-0.02, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))
# axis(side = 4, lwd.ticks = 2, tck=-0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 0.3, 0.6))
# axis(side = 1, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#      at = c(0, 100, 200, 300, 400)) 
# axis(side = 2, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#     at = c(0, 0.3, 0.6))
# axis(side = 3, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1,
#      at = c(0, 100, 200, 300, 400))     
# axis(side = 4, lwd.ticks = 2, tck=0.01, labels = F,  cex.axis = 2, las = 1, 
#    at = c(0, 0.3, 0.6))
# 
# box(lwd = 2)
# 
# mtext("BGE", side = 2, line = 4, cex = 1.25)
# mtext("Distance (m)", side = 1, line = 4, cex = 1.25)
# 
# # Simple linear regression for BGE
# BGE.fit <- lm(metab$BGE ~ metab$dist)
# BGE.R2 <- round(summary(BGE.fit)$r.squared, 2)
# BGE.int <- BGE.fit$coefficients[1]
# BGE.slp <- BGE.fit$coefficients[2]
# clip(0, 375, 0, 0.58)
# abline(a = BGE.int, b = BGE.slp, col = "black", lwd = 2.5, lty = 6)
# text(40, 0.535, labels = bquote(italic(R)^2 == .(BGE.R2)), cex = 1.5)
# 
# dev.off() # this writes plot to folder
# graphics.off() # shuts down open devices
# par(opar)
```

```{r fig.width=4, fig.height=5,echo=FALSE,fig.cap="Microbial Processes Along Gradient"}
# img <- readPNG("../figures/Figure1.png")
# grid.raster(img)
```


